### ISSUE TITLE
[BUG/ISSUE] Negative scale factor in CH4 simulation

### ISSUE THREAD

**Issue by subtlescreamer** [geoschem/geos-chem#981]
Shout-out to the community and the geos-chem team for all their extensive documentation and descriptive wikis. 

### Description of the problem
I am working to better understand Methane concentrations from the CLIMCAPS retrieval algorithm by comparing them to Methane simulations using Geos-chem. 
Currently running a debug test simulation for Methane and have come across an issue - When the simulation starts, I am getting a Negative scale factor (negative concentrations) from the first time-step, and the simulation ends within a couple of minutes pointing to errors in the files - HCO_Run, HCOI_GC_Run and Emissions_Run.

### Description of troubleshooting performed
1. Checked all configurations in input.geos and HEMCO_Confic.rc - seems that the environment has been set up properly.
2. Tried remaking geos rundirs and re-doing the whole process again.
3. Doesn't seem to be a compiler/memory issue, as previous standard simulations ran fine.

### GEOS-Chem version
12.9.3

### Set of files included - 
1. Input.geos file.
2. HEMCO_Config.rc file.
3. A log file of my test simulation - log.bench. 
4. A log file for HEMCO - HEMCO.log.

Can somebody tell me where I'm going wrong or help me pin-point the source of this error?

[log.bench.txt](https://github.com/geoschem/geos-chem/files/7444448/log.bench.txt)
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/7444449/HEMCO_Config.rc.txt)
[input.geos.txt](https://github.com/geoschem/geos-chem/files/7444450/input.geos.txt)
Note - I couldn't attach the particular restart file and my lastbuild file here because of github's format issues. If need be, I can send those over too!
[HEMCO.log](https://github.com/geoschem/geos-chem/files/7447019/HEMCO.log)

### Software versions
- Compilers (Intel or GNU, and version): Intel, 16-core processor.

**Comment by yantosca** [comment]
Thanks for writing @subtlescreamer.  Could you also post your HEMCO.log file?  I believe this is an error in HEMCO, and it should show up in HEMCO.log.  (Rename it to HEMCO.log.txt so that it'll post here.)

**Comment by subtlescreamer** [comment]
> @subtlescreamer I noticed a large amount of space (maybe a tab) between the "emi" and "_ch4" in this section of HEMCO_Config.rc. Wonder if that is causing an issue.
> 
> ```
> ### Other Anthro (industry, stationary combustion, mobile combustion, aircraft, composting, field burning of agri. residues, etc.) ###
> 0 CH4_OTHER__1A1_1B1_1B2 $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A1b_1A1c_1A5b1_1B1b_1B2a5_1B2a6_1B2b5_2C1b.generic.01x01.nc emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A1a        $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A1a.generic.01x01.nc                                        emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A2         $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A2.generic.01x01.nc                                         emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A3a_CDS    $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A3a_CDS.generic.01x01.nc                                    emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A3a_CRS    $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A3a_CRS.generic.01x01.nc                                    emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A3a_LTO    $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A3a_LTO.generic.01x01.nc                                    emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A3a_SPS    $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A3a_SPS.generic.01x01.nc                                    emi                                                                                                                                                    _ch4 1970-2003/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A3b        $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A3b.generic.01x01.nc                                        emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A3c_1A3e   $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A3c_1A3e.generic.01x01.nc                                   emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A3d_1C2    $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A3d_1C2.generic.01x01.nc                                    emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__1A4         $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_1A4.generic.01x01.nc                                         emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__6B          $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_2B.generic.01x01.nc                                          emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__2C          $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_2C1a_2C1c_2C1d_2C1e_2C1f_2C2.generic.01x01.nc                emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__4F          $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_4F.generic.01x01.nc                                          emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> 0 CH4_OTHER__6C          $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_6C.generic.01x01.nc                                          emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  8 1
> #0 CH4_OTHER__7A         $ROOT/CH4/v2017-10/EDGARv432/EDGAR_v432_CH4_IPCC_7A.generic.01x01.nc                                          emi                                                                                                                                                    _ch4 1970-2012/1/1/0 C xy kg/m2/s     CH4 -  9 1
> )))EDGARv432
> ```

**Comment by yantosca** [comment]
@subtlescreamer, I wasn't able to tell if the run that you did was an "out-of-the-box" run (i.e. with no modifications), or if you added any modifications (i.e. turning on non-default emissions).  Could you please clarify?

If you haven't done an "out-of-the-box" run, then I would try that...just create a new rundir and set it up to run for 1 day, without touching any of the other defaults.  If the run finishes without errors, then you can look at the configuration files between the "out-of-the-box" run and the run that failed to try to get an idea of where the error is coming in.

**Comment by yantosca** [comment]
@subtlescreamer, I was just able to run successfully for 20160101 without the error using the 13.2.1 code, which is the latest version.  Could you try with this version?  I wonder if something got fixed since 12.9.3.

**Comment by yantosca** [comment]
@subtlescreamer, I believe that you cannot flush negative values to 0 in HEMCO for the CH4 simulation.  This is because some of the surface flux quantities have + and - values.  The out-of-the-box HEMCO_Config.rc for the CH4 simulation uses:
```
Negative values:             2
```
[From the HEMCO User's Guide](http://wiki.seas.harvard.edu/geos-chem/index.php/The_HEMCO_User%27s_Guide):

| Setting | Default | Explanation |
|-------- |-------- |----------- |
| Negative values | 0 | Integer value defining how negative values are handled (0 = no  negative values are allowed; 1 = all negative values are set to zero and  a warning is prompted; 2 = negative values are kept as they are). |

**Comment by subtlescreamer** [comment]
> @subtlescreamer, I believe that you cannot flush negative values to 0 in HEMCO for the CH4 simulation. This is because some of the surface flux quantities have + and - values. The out-of-the-box HEMCO_Config.rc for the CH4 simulation uses:
> 
> ```
> Negative values:             2
> ```
> 
> [From the HEMCO User's Guide](http://wiki.seas.harvard.edu/geos-chem/index.php/The_HEMCO_User%27s_Guide):
> 
> Setting	Default	Explanation
> Negative values	0	Integer value defining how negative values are handled (0 = no negative values are allowed; 1 = all negative values are set to zero and a warning is prompted; 2 = negative values are kept as they are).

**Comment by subtlescreamer** [comment]
@yantosca I set up GC-13.2.1 and the test simulation ran successfully. Thanks for your suggestion.

Regarding the negative value setting, if I set the value to 2, the simulation would run fine but my SpeciesConc_CH4 would have negative values since the first date of the simulation. I guess there was an issues either with my restart file or the way I set up my environment. 
Nevertheless, let's hope my modified emissions scenario runs - will keep you updated as I proceed.
